core/modules/widgets/select.js

```diff
@@ -53,6 +53,17 @@ Handle a change event
 SelectWidget.prototype.handleChangeEvent = function(event) {
 	var value = this.getSelectDomNode().value;
 	this.wiki.setText(this.selectTitle,this.selectField,this.selectIndex,value);
+
+	this.activeChild = 2*(this.getSelectDomNode().selectedIndex)+1;
```

Comment:
This is here becuase for some reason every other element of this.parseTreeNode.children is empty starting with index 0 so there for a select widget with n options there are 2n+1 children. This gives the index of the child that corresponds to the option picked. If that changes than this won't work anymore.
The reason I am doing it this way is because before when I had it searching through the values of each child to match this.getSelectDomNode().value it would only work for simple options, not for any that had value=foo parts to them.

If there is a danger that the children will be organized differently in the future than I will try to find a way to make the previous method work.